akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -386,6 +388,15 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](xs: Array[B], start: Int, len: Int): Int = {
+      // min of the bytes available top copy, bytes there is room for in dest and the requested number of bytes
+      val bytesToCopy = length.min(len).min(xs.length - start)
+      if (bytesToCopy > 0) {
+        Array.copy(bytes, startIndex, xs, start, bytesToCopy)
+      }
+      bytesToCopy
```

Comment:
Is it defined what we should do when `start > xs.length`? I guess not.

Reply:
I'll look into what the stdlib does for that case.

Reply:
Oh, it just passes it on to System.arrayCopy so I guess no bounds check is fine here/will lead to the same exception

Reply:
It'd mean returning a negative value here, right? GIGO, I guess :)